src/js/components/productSupplier/create/subsections/BasicDetails.jsx

```diff
@@ -0,0 +1,197 @@
+import Switch from 'components/form-elements/v2/Switch';
+import TextInput from 'components/form-elements/v2/TextInput';
+import Subsection from 'components/Layout/v2/Subsection';
+import { INVENTORY_ITEM_URL } from 'consts/applicationUrls';
+import { debounceOrganizationsFetch, debounceProductsFetch } from 'utils/option-utils';
+
+const BasicDetails = ({ control, errors }) => {
+  const {
+    debounceTime,
+    minSearchLength,
+  } = useSelector((state) => ({
+    debounceTime: state.session.searchConfig.debounceTime,
+    minSearchLength: state.session.searchConfig.minSearchLength,
+  }));
+
+  // Watch product's input changes live, in order to display a "View Product" link
+  // with a proper product id
+  const product = useWatch({ control, name: 'product' });
+
+  return (
+    <Subsection
+      title={{ label: 'react.productSupplier.form.subsection.basicDetails', defaultMessage: 'Basic Details' }}
+      collapsable={false}
+    >
+      <div className="form-grid-3">
+        <Controller
+          name="code"
+          control={control}
+          render={({ field }) => (
+            <TextInput
+              title={{ id: 'react.productSupplier.form.code.title', defaultMessage: 'Source Code' }}
+              tooltip={{
+                id: 'react.productSupplier.form.code.tooltip',
+                defaultMessage: 'Unique code that identifies this record. ' +
+                  'Auto-generated based on the product and supplier codes, ' +
+                  'but can be manually overwritten',
+              }}
+              placeholder="Leave blank to autogenerate"
+              {...field}
+            />
+          )}
+        />
+        <Controller
+          name="product"
+          control={control}
+          render={({ field }) => (
+            <SelectField
+              title={{ id: 'react.productSupplier.form.product.title', defaultMessage: 'Product' }}
+              required
+              async
+              errorMessage={errors.product?.message}
+              button={product
+                ? {
+                  id: 'react.productSupplier.form.viewProduct.title',
+                  defaultMessage: 'View Product',
+                  onClick: (id) => window.open(INVENTORY_ITEM_URL.showStockCard(id), '_blank'),
+                }
+                : null}
+              loadOptions={debounceProductsFetch(debounceTime, minSearchLength)}
+              {...field}
+            />
+          )}
+        />
+        <Controller
+          name="legacyCode"
+          control={control}
+          render={({ field }) => (
+            <TextInput
+              title={{ id: 'react.productSupplier.form.legacyCode.title', defaultMessage: 'Legacy Code' }}
+              tooltip={{
+                id: 'react.productSupplier.form.legacyCode.tooltip',
+                defaultMessage: 'Reference to this record in a previous or parallel purchasing system',
+              }}
+              {...field}
+            />
+          )}
```

Comment:
Do you think we should move these `<Controller name="" render {...fields}` into `TextInput`, `SelectField`, `DateField` components?
The instead of 
```jsx
<Controller
          name="legacyCode"
          control={control}
          render={({ field }) => (
            <TextInput
              title={{ id: 'react.productSupplier.form.legacyCode.title', defaultMessage: 'Legacy Code' }}
              tooltip={{
                id: 'react.productSupplier.form.legacyCode.tooltip',
                defaultMessage: 'Reference to this record in a previous or parallel purchasing system',
              }}
              {...field}
            />
          )}
```

we would do
```jsx
          <TextInput
             name="legacyCode"
            control={control}
            title={{ id: 'react.productSupplier.form.legacyCode.title', defaultMessage: 'Legacy Code' }}
            tooltip={{
              id: 'react.productSupplier.form.legacyCode.tooltip',
              defaultMessage: 'Reference to this record in a previous or parallel purchasing system',
            }}

          />
```
cc @alannadolny @kchelstowski

Reply:
I get the point, but if it was up to me, I’d suggest sticking to calling the <Controller> with render for now, until we get used to the library and until we know any potential use cases we might have with the new inputs. Using <Controller> + render is not the only way to use the react-hook-form, hence if we added it to the inputs, any other approach would be impossible to use, or if we wanted to use input without the form, because it would always expect to receive the control prop.

Reply:
That sounds good, I see what you mean.
Let's not try coming up with "clever abstract" ideas and stick with simple implementation for now, in the future this can easily be improved with more cases

Reply:
I am not sure about that, because of that we will be limited to the use of the `Controller` so, in a case when we would like to use our inputs without the form library it can cause some issues. I would leave it as it is.